_posts/2023-17-10-remote-attestation-blog-part1.markdown

```diff
@@ -0,0 +1,137 @@
+
+This framework allows changes to the user’s computer, the attested (your remote servers), to be detected by authorized parties, the attestors (you on your local servers). In this framework, the attestor machine is already trusted by the customers, while the nodes they are provisioning through the cloud service providers are not. 
+
+Remote attestation can provide different services, such as measured boot attestation and runtime integrity monitoring, using a hardware-based cryptographic root of trust, otherwise known as a Trusted Platform Module (TPM). A TPM is a chip that conforms to the standards of a secure cryptoprocessor, which is a dedicated microcontroller designed to secure hardware through integrated cryptographic keys. It is used to store artifacts that work to authenticate the platform it is on. 
+
+
+<div style="text-align: center;">
+<img  src="/Users/margieruffin/Desktop/Research/Keylime/keylime.github.io/assets/images/RemoteAttestation.png" alt="Remote Attestation Diagram">
+</div>
+
+**Figure 1: Remote Attestation is used to provide trust between an untrusted party and a trusted party. You can use services like Measured Boot Attestation and Runtime Integrity Monitoring to prove systems are trustworthy.**
+
+In this article, we will detail the different aspects of remote attestation based on the following sections:
+<ul>
+    <li> What is Secure Boot? </li>
+    <li> What is Measured Boot? </li>
+    <li> Measured Boot and Secure Boot go Hand in Hand </li>
+    <li> Integrity Measurement Architecture (IMA) for Continuous Attestation</li>
+    <li> Introduction to Keylime, A Remote Attestation Software </li>
+</ul>
+
+<h3 style="text-align left; color:#6B98BF;"> What is Secure Boot? </h3>
+
+Trusted boot technologies rely on a “Root of Trust” (RoT), a source that can always be trusted within a cryptographic system. The RoT is usually a piece of code or hardware that’s been hardened well enough that it is unlikely to be compromised. The RoT cannot be modified at all or cannot be modified without cryptographic credentials. 
+
+Unified Extended Firmware Interface (UEFI) Secure Boot is a security measure developed to ensure that a device is booted using only software trusted by the Original Equipment Manufacturer (OEM). The goal is to prevent malicious software from being loaded and executed early in the boot process. During secure boot, the kernel checks if a cryptographic signature, a hash value, on the boot loader matches a pre-loaded value signed by the manufacturer, stored inside the BIOS, that anyone can check using a public key made available. 
```

Comment:
The kernel does not have directly something to do with Secure Boot because it is the last component in that chain. (In some cases the UEFI Secure Boot certificates are loaded onto the kernel keyring, but that is a more complex topic). The UEFI firmware does the verification in the first step, then the bootloaders can either use the firmware to check the next component or as in the case with most x86 UEFI Linux systems that use a shim (https://github.com/rhboot/shim) via the shim lock protocol (e.g. https://www.gnu.org/software/grub/manual/grub/html_node/UEFI-secure-boot-and-shim.html).  For simplification we can just say that in Secure Boot that the next component is verified by the component before.

On a normal UEFI implementation you have three major keys/certificates involved:
* Platform Key (PK): Is created by the OEM, allows the enrollment of the KEK
* Key Exchange Key (KEK): Is on most systems one from Microsoft. Allows to update i.e. the DB entries
* Allowed signatures database (DB): certificates actually used to verify the boot components. On x86 UEFI systems you likely find here the Windows CA and the third-party Windows CA.

Details can be found for example here: https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/windows-secure-boot-key-creation-and-management-guidance?view=windows-11

Most users only will not interact with those keys, but in many cases you can also enroll your own setup. I don't know what is most common by the cloud providers.